src/com/google/cloud/datastore/types.js

```diff
@@ -262,9 +260,9 @@ foam.CLASS({
 
       var name = new Array(ids.length);
       for ( var i = 0; i < ids.length; i++ ) {
-        name = ids[i] + ( i < ids.length - 1 ? ':' : '' );
+        name[i] = this[ids[i]];
       }
-      return { kind: this.getOwnDatastoreKind(), name: name };
+      return { kind: this.getOwnDatastoreKind(), name: name.join('_') };
```

Comment:
Why is this for loop needed? Would this be the same as just:

return { kind: this.getOwnDatastoreKind(), name: this.id.join('_') };

Since 'id' is set to an array where each element is a part of the multi-part key, just like you were creating.
Also, couldn't you just use foam.json.stringify(this.id) to avoid issues with keys containing either _ or :?